src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -674,12 +693,25 @@ public static void onWorldLoad(@NotNull final World world)
                     MineColonies.getLogger().error("Failed to save " + FILENAME_MINECOLONIES + " backup!");
                 }
 
+                //load the structures when we know where the world is
+                Structures.init();
+
                 @NotNull final File file = getSaveLocation();
                 @Nullable final NBTTagCompound data = loadNBTFromPath(file);
                 if (data != null)
                 {
                     readFromNBT(data);
                 }
+                if (serverUUID == null)
+                {
+                    serverUUID = UUID.randomUUID();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Synchronize this lazy initialization of 'serverUUID' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2444)